docs/extensions/garden-api-access.md

```diff
@@ -129,8 +129,61 @@ With this, extensions are restricted to work with objects in the garden cluster
-Note that this is done outside of Gardener and might require an additional controller that manages RBAC for extension clients in the garden cluster.
+Let's consider the following example: An extension requires the privileges to create [`authorization.k8s.io/v1.SubjectAccessReview`](https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/subject-access-review-v1/)s (which is not covered by the "default" permissions mentioned above).
+This requires a human Gardener operator to create a `ClusterRole` in the garden cluster with the needed rules:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: extension-create-subjectaccessreviews
+  annotations:
+    authorization.gardener.cloud/extensions-serviceaccount-selector: '{"matchLabels":{"controllerregistration.core.gardener.cloud/name":"<extension-name>"}}'
+  labels:
+    authorization.gardener.cloud/custom-extensions-permissions: "true"
+rules:
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - create
+```
+
+Note the label `authorization.gardener.cloud/extensions-serviceaccount-selector` which contains a label selector for `ServiceAccount`s.
+
+There is a controller part of `gardener-controller-manager` which takes care of maintaining the respective `ClusterRoleBinding` resources.
+It binds all `ServiceAccount`s in the seed namespaces in the garden cluster (i.e., all extension clients) whose labels match.
+You can read more about this controller [here](../concepts/controller-manager.md#-extension-clusterrole--reconciler).
+
+### Custom Permissions
+
+If an extension wants to create a dedicated `ServiceAccount` for accessing the garden cluster **without** automatically inheriting all permissions of the gardenlet, it first needs to create a garden access secret in its extension namespace in the seed cluster:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-custom-component
+  namespace: <extension-namespace>
+  labels:
+    resources.gardener.cloud/purpose: token-requestor
+    resources.gardener.cloud/class: garden
+  annotations:
+    serviceaccount.resources.gardener.cloud/name: my-custom-component-extension-foo
+    serviceaccount.resources.gardener.cloud/labels: '{"foo":"bar}'
+type: Opaque
+```
+
+❗️**️Do not prefix the service account name with `extension-` to prevent inheriting the gardenlet permissions!** It is still recommended to add the extension name (e.g., as a suffix) for easier identification where this `ServiceAccount` comes from.
+
+Next, you can follow the same approach [described above](#additional-permissions).
+However, the `authorization.gardener.cloud/extensions-serviceaccount-selector` annotation should **not** contain `controllerregistration.core.gardener.cloud/name=<extension-name>` but rather custom labels, e.g. `foo=bar`.
+
+This way, the created `ServiceAccount` will only get the permissions of above `ClusterRole` and nothing else.
```

Comment:
> With "above" you mean the `ClusterRole` in the garden cluster, right? 

Yep 😄 

> I was slightly confused because I see a `Secret` above and not a `ClusterRole`.

If you scroll up ("more above") then you should see the `ClusterRole`. But this is actually part of another section. What do you recommend to make it clear - remove the word "above"? Or link to the other section?

Reply:
I would prefer to link to the section with the `ClusterRole`, e.g.

```
This way, the created `ServiceAccount` will only get the permissions of [above `ClusterRole`](#additional-permissions) and nothing else.
```